src/core/EnvelopeAndLfoParameters.cpp

```diff
@@ -400,23 +408,20 @@ void EnvelopeAndLfoParameters::updateSampleVars()
 
 	const float frames_per_env_seg = SECS_PER_ENV_SEGMENT *
 				Engine::mixer()->processingSampleRate();
-
 	// TODO: Remove the expKnobVals, time should be linear
 	const f_cnt_t predelay_frames = static_cast<f_cnt_t>(
 							frames_per_env_seg *
 					expKnobVal( m_predelayModel.value() ) );
 
-	const f_cnt_t attack_frames = qMax( minimumFrames,
-					static_cast<f_cnt_t>( frames_per_env_seg *
-					expKnobVal( m_attackModel.value() ) ) );
+	const f_cnt_t attack_frames = static_cast<f_cnt_t>( frames_per_env_seg *
+					expKnobVal( m_attackModel.value() ) );
 
```

Comment:
You've pushed something like this before and the discussion here is still valid:
https://github.com/LMMS/lmms/pull/5544#discussion_r444347291

Reply:
That is not added by me
In this PR only parameters in the envelope definition is changes, nothing else
Why git marked this as added, i have no idea.
I use _Notepad++_ as editor.
Content of code:

Reply:
You are working from your master and your master branch has your stuff on top of it. It isn't 'clean'. Also it looks like you are still working from the github web interface and not pushing your changes from your computer directly.

Reply:
> Also it looks like you are still working from the github web interface and not pushing your changes from your computer directly.

That is correct. I add the files on the git-UI.
Why would that be a problem?
I have washed all part of the repo away (its *archived* as of now, but i will propl.  delete later)
I plan to fork current master, and then to add the 7 lines of edited code
That is impossible to botch.. isent it..?

Reply:
>  Why would that be a problem?

This is how you do it: https://github.com/LMMS/lmms/wiki/Contributing-Step-by-Step
Especially the part on submitting a patch: https://github.com/LMMS/lmms/wiki/Submitting-a-patch

Have you set up your remotes? (origin, upstrem etc.)
Run this command anywhere in your lmms code tree:
git remote --verbose

Reply:
Thank you Oscar, but that creates a problem.
I have made 7 new features. I stupidly thought that would be a suitable sized PR. 
I asked DOM on Discord, if i could add each of the codeblocks to my gitRepo-source-code, and in turn do one PR.
That would be possible, since the the blocks were complete discrete -Eg i have not changed existing *code*, only parameters and added completely new functions( + the necessary UI). 
So the plan was to add one feature, have it reviewed, and then add the next, directly into my gitHub, using the page-functionality for exactly *that*:
![image](https://user-images.githubusercontent.com/6662066/95766294-559d8d00-0cb3-11eb-9761-1032e7dd7f5e.png)

Otherwise a will have to roll back my local LMMS-code to a state before the second added feature, and then do the commit through git (as you describe) from my local mint in Virtual-box. -I practically need to redo my existing work.
Unless it is a fail on gitHub, and then i should be telling them about it, i cant understand why adding code directly -eg:

Reply:
>  I will take this up with gitHub, if they are willing to - At least i would get an understanding for why this is not working.

Please don't bump GitHub about this, I'm sure they have better things to do.


> Edit-file > Commit
> Should lead to _any_ issues at all. The changed code should appear as '**+**' in the compare, and the previous as '**-**'.
> Absolutely _nothing else_ should be marked.

I'm pretty sure the changes you are about to commit are visible somewhere and you've just missed it.

I think the problems you are seeing here is threefold.
1. You are working from the master branch. Always make a separate branch for the new changes and keep the lmms branches (master, stable-1.2 etc.) clean.
2. Your master branch isn't the same as lmms/master.
3. You don't make commits from a branch but add the files you are working on via the online editor. This makes it hard (at least for me) to follow what your doing. If you

Points 1, 2 and 3 above are all solved by following the instructions on our wiki page. Maybe you have a method that actually is legitimate but the PR's you have made so far suggest that this isn't the case. I will not be able to assist you here any further unfortunately as you don't follow a method that I have any deeper knowledge in. Sorry!

Reply:
> I will not be able to assist you here any further unfortunately as you don't follow a method that I have any deeper knowledge in

Reply:
@musikBear as @zonkmachine has explained, you should create a new branch and do each PR from there.  
![image](https://user-images.githubusercontent.com/6345473/96286675-c04e1180-0fae-11eb-8ec2-016ef6cc7ad1.png)

Once you create your branch, you can use the Git-GUI to switch to it.  Your stuff on master won't be lost, and then you can bring over what you need.  Sometimes, this is as easy as using a friendly command `git cherry-pick`, other times it's easier to just repeat the changes again.

Once you have the branch ready, you can open a brand new PR without the other cruft.  Best wishes and thanks for working hard on this, source code management is arguably tricker than the code sometimes and we're sympathetic to that.   Personally, I started with the Git-GUI too, but found myself unhappy with the results and took a few weeks to learn the command line. <3

Reply:
> help you with this **if needed**.

I have like 10 jokes pending.. Guess *"If the pope is catholic, its needed"* would do.
I will definitively need help, but i do have a plan
I have already made my current repo into a archived-repo
That should make it possible to start from scatch(?)
I therefore plan to make a brand new copy of *newest Master* as my *new* own Master.
I have gitDesktop installed.
I ought to be able to 
* take one single file, and commit that to my new-master.
* create a PR from that new Master, that now should include **one** new file
Ask for review of *that* change.
Does that sound as a functional plan?
I dont understand why there are *options* for adding *new* files *directly* in **github.com**, and then commit these changes, again with the tools on github.com, where there simply is a button labelled **Commit**, when it messes everything up.
I have asked on gitHub forum, but the question remains unanswered.
First i need to make a new **Own**-Master based on current. I am reading up on that, atm :)
I will come back to you, if i get stuck I have never used gitDesktop, so i have a steep climb, i am afraid -Much Appreciated! 👍

Reply:
> I dont understand why there are options for adding new files directly in github.com, and then commit these changes, again with the tools on github.com, where there simply is a button labelled Commit, when it messes everything up.

GitHub's website functionality is a convenience and isn't really intended as a full replacement to a code editor (at least not yet, this will likely get better in the future).   But that's just fine, the website commits can be squashed like any others.  That's not the problem.  The problem is when you make a new feature branch, it copies the stuff you did to `master`.  So leave `master` alone.  Easy right?  No, we all edit `master` because it's the default branch we clone, so this problem is very common.  It's not unique to git nor unique to GitHub, but I'll digress.  Leave `master` alone.

> create a PR from that new Master, that now should include one new file Ask for review of that change.

No, don't edit `master`.  Reset it to match `lmms/lmms` and leave it that way.  Only fast forward it once in a while.  Do your changes on the branch by selecting it from GitHub's dropdown menu (or better, switching your branch locally and using your PC to edit the changes). <3

>  > help you with this if needed.

> I have like 10 jokes pending.. Guess "If the pope is catholic, its needed" would do.

Sorry, I don't understand the joke, but my offer still stands.  The proverb I was raised with... 

> *"Give a man a fish and you feed him for a day. Teach a man to fish and you feed him for a lifetime."*

Let's 🎣